### PR TITLE
Add Graph3 (add DFS in Iterative Way)

### PR DESCRIPTION
**Pull request by dongjji**
# Welcome to the JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?
- [o] New Implementation of Data Structure

### Checklist:

- [o] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [o] This pull request is all my own work -- I have not plagiarized.
- [o] I know that pull requests will not be merged if they fail the automated tests.
- [o] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [o] All new JavaScript files are placed inside an existing directory.
- [o] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [o] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [o] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by dongjji** [comment]
I have my first time contributing Open Source Project. 
Is it OK what I did?

File: Data-Structures/Graph/Graph3.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          visited[neighbor] = true
+          queue.push(neighbor)
+        }
+      })
+    }
+    return result
+  }
+}
+
+// example
```
Comment (raklaptudirm) [inline Data-Structures/Graph/Graph3.js:114]: Remove these examples.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Write unit tests for your algorithm in `jest`.

**Comment by dongjji** [comment]
@raklaptudirm 
I apologize to disturb you, Is ther anything to do more to get permission for merge?

File: Data-Structures/Graph/Graph3.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  addVertex (vertex) {
+    if (!this.adjacencyObject[vertex]) this.adjacencyObject[vertex] = []
+  }
+
+  addEdge (vertex1, vertex2) {
+    if (this.adjacencyObject[vertex1] && this.adjacencyObject[vertex2]) {
+      this.adjacencyObject[vertex1].push(vertex2)
+      this.adjacencyObject[vertex2].push(vertex1)
+    } else {
+      return undefined
```
Comment (raklaptudirm) [inline Data-Structures/Graph/Graph3.js:15]: This statement is redundant, as the function will return undefined by default.

File: Data-Structures/Graph/Graph3.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  removeEdge (vertex1, vertex2) {
+    if (this.adjacencyObject[vertex1] && this.adjacencyObject[vertex2]) {
+      this.adjacencyObject[vertex1] = this.adjacencyObject[vertex1].filter(
+        (v) => v !== vertex2
+      )
+      this.adjacencyObject[vertex2] = this.adjacencyObject[vertex2].filter(
+        (v) => v !== vertex1
+      )
+    } else {
+      return undefined
```
Comment (raklaptudirm) [inline Data-Structures/Graph/Graph3.js:28]: Same comment.

File: Data-Structures/Graph/Graph3.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  removeVertex (vertex) {
+    if (this.adjacencyObject[vertex]) {
+      while (this.adjacencyObject[vertex].length) {
+        const adjacentVertex = this.adjacencyObject[vertex].pop()
+        this.removeEdge(vertex, adjacentVertex)
+      }
+    } else {
+      return undefined
```
Comment (raklaptudirm) [inline Data-Structures/Graph/Graph3.js:39]: Same comment.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Data-Structures/Graph/Graph3.js (hunk 4 above)
Comment (dongjji) [reply Data-Structures/Graph/Graph3.js]: @raklaptudirm 
I modified the code what you commented.
Thanks a lot raklaptudirm.
Thanks to you I learn a lot of things to contribute open source project.

**Review by raklaptudirm — APPROVED** [review]
Welcome to the community!

**Pull request merged**